Add admin app (#1733)
bonitasoft/bonita-doc: modules/ROOT/pages/admin-application-overview.adoc

= Bonita Administrator Application overview

> :description: This page describes the Bonita Administrator Application: its values, how to access it, and how to customize it. +

Reviewer: ```suggestion
:description: This page describes Bonita Administrator Application: its values, how to access it, and how to customize it. +
```

{description}

== Value

> The Bonita Administrator Application allows anyone with the _Administrator_ profile to manage the use of Bonita throughout a department or a company. +

Reviewer: ```suggestion
Bonita Administrator Application allows anyone with the _Administrator_ profile to manage the use of Bonita throughout a department or a company. +
```

Alternatively, you can deploy an application by importing all its components in the following order:

* Organization then profiles
* API extensions then pages, layout, themes
* Business Data Model then processes
* Application descriptor

Once deployed, log into Bonita and navigate to the application through the Application Directory.

== Create your own Administrator application from ours

Creating your adminstrator application based on Bonita Administrator Application means that you can change some of its existing pages, its navigation, its logo, its layout and/or its graphical theme and make those changes visible in the application descriptor. +

=== Import it in Bonita Studio

From Bonita Studio *Welcome page* _Resources_ tile, click on "Import Bonita Admin Application".

image::images/UI2021.1/studio-get-application.png[Get application from Studio]

You can also get it from our https://github.com/bonitasoft/bonita-admin-application/[Admin Application GitHub repository]. +
There are two ways to get the .bos file. You can either get it from the _Releases_ on the right or by cloning the repository of your choice and building it using maven. +

> Building the repository will create a .bos file in the `target` folder. You can import this one into the development environment. +

Reviewer: ```suggestion
Building the repository will create a .bos file in the `target` folder. You can then import it into the development environment. +
```

Using this method is also useful if you want to get an older version of an application or if you need a weekly version with a potential bugfix. +
After validation, the application is imported or cloned; you can view its content in the project explorer, deploy it, and open it. +

=== Change the existing pages

. In the studio's Project Explorer, select the page to customize.

>  * All pages previously created using Google Web Toolkit framework have been recreated with Bonita UI Designer. They can be customized in the UI Designer. They come with the Administrator Application you have download from Bonita Studio. You will see them in the Project Explorer.

Reviewer: ```suggestion
 * All pages previously created using Google Web Toolkit framework have been recreated with Bonita UI Designer. They can be customized in the UI Designer. They come with the Administrator Application you have downloaded from Bonita Studio. You will see them in the Project Explorer.
```

 * All pages previously created in other, more recent technologies, have been wrapped as _custom pages_. They cannot be customized in the UI Designer. Since they cannot be imported in a .bos archive, they do not come with the Administrator Application you have downlaoded. Instead, they come as provided resources. You can find them in the "Resources" tab of the application.
. Double-click on the page. It opens in the UI Designer. You can start editing the page.
. Save it when you are done. Save it as another page if you need to keep Bonita's original page for further use. 

Saving in the UI Designer makes the new page available in Bonita Studio.

> === Change its navigation menu

Reviewer: To be consistent with the previous title:
```suggestion
=== Change the navigation menu
```

. Get all the application's pages available:
 * If you have created new pages in the UI Designer and saved them, they are ready to be used in an application menu. 
 * If you have created new pages in another IDE, go to the *Resources* menu option of the Administrator application, and Install your pages.

>    At deployment time, since these external resources are not part of the project, make sure you also deploy them (CAN THEY BE ATTACHED AS DEPENDENCIES IN THE MAVENIZED PROJECT IN 2021.2, OR ADDED TO THE GIT REPO IN ANOTHER WAY?)

Reviewer: No they still can't
```suggestion
   At deployment time, since these external resources are not part of the project, make sure you also deploy them.
```

. In Bonita Studio, go to the "Project Explorer" on the left and double click on the "bonita-admin-application.xml" application descriptor for Community, or "bonita-admin-application-sp.xml" for Enterprise.
. In the editor, change the order in the menu, remove the pages you do not need, add other Bonita provided pages, and/or add yours, either in the menu or as orphan pages. +

>   Orphan pages are not visible in the menu, but accessed through the navigation from another page, like a case details page is accessed through a case list.

Reviewer: ```suggestion
  Orphan pages are not visible in the menu, but can be accessed through the navigation from another page, like a case details page is accessed through a case list.
```

[NOTE]
====

So far, the Bonita Administrator Application is mostly thought for desktop and laptop screen sizes, as we believe this is the most likely screen sizes to be used to administrate Bonita. If you target mobile usage, you may need to rework the widgets display in the UI Designer in the application pages, but Bonita default layout is made to adapt to a mobile screen size.  +
====

. Create a new xref:layout-development.adoc[layout] and make it available like a page (UI Designer or *Resources* if created in another IDE)
. In the *Look & Feel* section of the application descriptor, clear the current layout content
. Start typing the name of the layout
. Select the new one from the autocomplete list
. Save the application descriptor

The new layout is applied.

=== Change its theme

. Create a new xref:customize-living-application-theme.adoc[theme] 
. Go to the Administrator application > *Resources*
. Install the new theme
. Go back to the application descriptor
. In the *Look & Feel* section, clear the current theme content
. Start typing the name of the theme
. Select the new one from the autocomplete list
. Save the application descriptor

The new layout is applied.

> At deployment time, since these external resources are not part of the project, make sure you also deploy them (CAN THEY BE ATTACHED AS DEPENDENCIES IN THE MAVENIZED PROJECT IN 2021.2, OR ADDED TO THE GIT REPO IN ANOTHER WAY?)

Reviewer: ```suggestion
At deployment time, since these external resources are not part of the project, make sure you also deploy them.
```
